pandas/parser.pyx

```diff
@@ -621,8 +621,10 @@ cdef class TextReader:
                 if isinstance(source, basestring) or PY3:
                     source = bz2.BZ2File(source, 'rb')
                 else:
-                    raise ValueError('Python 2 cannot read bz2 from open file '
-                                     'handle')
+                    content = source.read()
+                    source.close()
+                    from pandas.compat import StringIO
```

Comment:
Fixed those two issues. Was a nice opportunity for me to [try out `--fixup`](http://fle.github.io/git-tip-keep-your-branch-clean-with-fixup-and-autosquash.html). We can do the `--autosquash` when the PR looks ready to go.